ext/cgimap/cgimap.cpp

```diff
@@ -339,6 +420,20 @@ boost::shared_ptr<output_buffer> hijack_rack_request::get_buffer_internal() {
 void hijack_rack_request::finish_internal() {
 }
 
+
+boost::posix_time::ptime hijack_rack_request::get_current_time() const {
+
+  return boost::posix_time::second_clock::local_time();
```

Comment:
this was a bit of a hack. time should have been stored in some private instance variable, and which is returned here, along with some set_current_time method. I think cgimap has something similar already.